redfish-core/lib/power_supply_metrics.hpp

```diff
@@ -294,83 +294,263 @@ inline void getValues(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                               std::vector<std::pair<
+                                   std::string, std::vector<std::string>>>>>&
+                               subtree) {
+        BMCWEB_LOG_DEBUG << "getValidInputHistory respHandler enter";
+
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR << "getValidInputHistory respHandler DBUS error: "
+                             << ec;
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        // Set the default value to resourceNotFound, and if we confirm finding
+        // association with chanssisID and powerSupplyID is correct, the error
+        // response will be cleared.
+        messages::resourceNotFound(asyncResp->res, "PowerSupplyMetrics",
+                                   "Metrics");
+
+        for (const auto& object : subtree)
+        {
+            // The association of this input history item is used to
+            // determine whether it belongs to this ChassisID, and therefore
+            // is valid to include for this power supply.
+            crow::connections::systemBus->async_method_call(
+                [callback{std::move(callback)}, asyncResp, chassisID,
+                 powerSupplyID, object](
+                    const boost::system::error_code ec,
+                    const std::variant<std::vector<std::string>>& endpoints) {
+                    if (ec)
                     {
-                        BMCWEB_LOG_ERROR << "Not a valid chassis ID:"
-                                         << chassisID;
-                        messages::resourceNotFound(asyncResp->res, "Chassis",
-                                                   chassisID);
+                        if (ec.value() == EBADR)
+                        {
+                            // This input history item has no chassis
+                            // association.
+                            BMCWEB_LOG_DEBUG
+                                << "This input history item has no chassis {"
+                                << chassisID << "} association";
+                            return;
+                        }
+
+                        BMCWEB_LOG_ERROR << "DBUS response error";
+                        messages::internalError(asyncResp->res);
                         return;
                     }
 
-                    if (chassisID != "chassis")
+                    const std::vector<std::string>* itemChassis =
+                        std::get_if<std::vector<std::string>>(&(endpoints));
+
+                    if (itemChassis != nullptr)
                     {
-                        BMCWEB_LOG_ERROR << "No Metrics for chassis ID:"
-                                         << chassisID;
-                        messages::resourceNotFound(asyncResp->res, "Chassis",
-                                                   chassisID);
-                        return;
-                    }
+                        // more stuff to do.
```

Comment:
Nit: I'm not sure this comment is helping.

Reply:
Hah! I apparently had a blank block of code in here at one point, filled it in, and forgot to remove the comment.